### PR TITLE
chore: update styleguide with info about composition api

### LINKED ISSUE #4128: Add composition API as required contributon practice
### Is your feature request related to a problem?

>Some notes to help us migrate to vue3/nuxt3:
> We need to reduce using `nuxt-property-decorator` first and use composition API instead. 
> It will help us to migrate to vue3/nuxt3
https://github.com/nuxt-community/nuxt-property-decorator/issues/83#issuecomment-1245139696. at least for newer components
related: https://github.com/kodadot/nft-gallery/issues/2893

_Originally posted by @preschian in https://github.com/kodadot/nft-gallery/issues/2690#issuecomment-1280409802_
      

### The task

- [ ] Edit style guide https://github.com/kodadot/nft-gallery/blob/main/STYLE_GUIDE.md
- [ ] Add examples how should components looks like
- [ ] Best we should follow [Vue3 style guide](https://vuejs.org/style-guide/rules-essential.html)

### PR DESCRIPTION
**Pull request by kkukelka**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [x] Chore

### What's new?

- [x] PR closes #4128
- [ ] Requires deployment <rubick/magick_issue>
- [x] Added best practice Composition API Examples and links to further reading

### Before submitting Pull Request, please make sure:

- [ ] My contribution builds **clean without any errors or warnings**
- [ ] I've merged recent default branch -- **main** and I've no conflicts
- [ ] I've tried to respect high code quality standards
- [ ] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://kodadot.xyz/transfer/?target=EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Review by prachi00 — APPROVED** [review]

File: STYLE_GUIDE.md [hunk 1]
```diff
@@ -90,6 +132,21 @@ Use shorthands for vue attributes
...
+  queryName: 'buyEventByProfile',
+  variables: {
+    id: address,
+  },
+})
+```
+For reference you can take a look at `useCarousel.ts` and its usage throughout the app. It will show you how to best abstract such calls into its own [composables](https://vuejs.org/guide/reusability/composables.html), which is one of the core concepts behind the Composition API.
+
+
+#### Class API (DEPRECATED)
```
Comment (vikiival) [inline STYLE_GUIDE.md:149]: I would personally remove this

**Review by vikiival — COMMENTED** [review]
I would pefsonally remove the class syntax in general

File: STYLE_GUIDE.md (hunk 1 above)
Comment (kkukelka) [reply STYLE_GUIDE.md]: done

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 50 usd

**Pull request merged**